aep/general/0203/aep.md.j2

```diff
@@ -1,5 +1,278 @@
+the value matches, but **should** error with `INVALID_ARGUMENT` if a change is
+requested.
+
+Potential use cases for immutable fields (this is not an exhaustive list) are:
+
+- Attributes of resources that are not modifiable for the lifetime of the
+  application (e.g. a disk type).
+
+**Note:** Fields which are "conditionally immutable" **must not** be given the
+immutable annotation.
+
+### Input only
+
+The use of `INPUT_ONLY` indicates that the field is provided in requests and
+that the corresponding field will not be included in output.
+
+Additionally, a field **should** only be described as input only if it is a
+field in a resource message or a field of a message included within a resource
+message. Notably, fields in request messages (a message which only ever acts as
+an argument to an RPC, with a name usually ending in `Request`) **should not**
+be described as input only because this is already implied.
+
+Potential use cases for input only fields (this is not an exhaustive list) are:
+
+- The `ttl` field as described in AIP-214.
+
+**Warning:** Input only fields are rare and should be considered carefully
+before use.
+
+### Optional
+
+The use of `OPTIONAL` indicates that a field is not required.
+
+A field **may** be described as optional if it is a field on a request message
+(a message that is an argument to an RPC, usually ending in `Request`), or a
+field on a submessage.
+
+### Output only
+
+The use of `OUTPUT_ONLY` indicates that the field is provided in responses, but
+that including the field in a message in a request does nothing (the server
+**must** clear out any value in this field and **must not** throw an error as a
+result of the presence of a value in this field on input). Similarly, services
+**must** ignore the presence of output only fields in update field masks (see:
+AIP-161).
+
+Additionally, a field **should** only be described as output only if it is a
+field in a resource message, or a field of a message farther down the tree.
+Notably, fields in response messages (a message which only ever acts as a
+return value to an RPC, usually ending in `Response`) **should not** be
+described as output only because this is already implied.
+
+Output only fields **may** be set to empty values if appropriate to the API.
+
+Potential use cases for output only fields (this is not an exhaustive list)
+are:
+
+- Create or update timestamps.
+- Derived or structured information based on original user input.
+- Properties of a resource assigned by the service which can not be altered.
+
+### Required
+
+The use of `REQUIRED` indicates that the field **must** be present (and set to
+a non-empty value) on the request or resource.
+
+A field **should** only be described as required if _either_:
+
+- It is a field on a resource that a user provides somewhere as input. In this
+  case, the resource is only valid if a "truthy" value is _stored_.
+  - When [creating](./standard-methods) the resource, a value **must** be
```

Comment:
nit: I think this type of enumeration should belong in the standard methods - it's a big backwards to do this type of cross-referencing about a specific standard method back to a lower-level AEP around behavior.

Reply:
I agree with this. I'll make an issue for this.